pkg/hubble/observer/local_node_watcher.go

```diff
@@ -17,21 +17,25 @@ import (
 // LocalNodeWatcher populate Hubble flows local node related fields (currently
 // only labels).
 type LocalNodeWatcher struct {
-	mu      lock.Mutex
-	updated int // update counter
-	cache   struct {
+	mu    lock.Mutex
+	cache struct {
 		// labels are represented as a Go map in node.LocalNode, but we need a
 		// key=val slice for Hubble flows.
 		labels []string
 	}
+	// init is used as a condition, it is closed the first time the
+	// LocalNodeWatcher gets an update from the localNodeStream.
+	init chan struct{}
 }
 
 // NewLocalNodeWatcher return a new LocalNodeWatcher. The given context control
 // whether the LocalNodeWatcher gets updated by the localNodeStream. It is safe
 // to use the returned LocalNodeWatcher once the context is cancelled, but its
 // information might be out-of-date.
 func NewLocalNodeWatcher(ctx context.Context, localNodeStream stream.Observable[node.LocalNode]) *LocalNodeWatcher {
-	watcher := LocalNodeWatcher{}
+	watcher := LocalNodeWatcher{
+		init: make(chan struct{}),
+	}
```

Comment:
I wonder if it would be possible to call `LocalNodeStore.Get()` here to block the creation of the this wrapper until the local node information is available. That way the callers could assume that the information is always here.

It would block Hubble until local node information is available, but I think that's fine

Reply:
Yes I thought about it and decided that the feature wasn't important enough to block the Hubble startup. My thinking was also that it shouldn't be designed around tests.

On the other hand in the current implementation it's easy to misuse the `LocalNodeWatcher`, so let's change it.